aexpect/remote.py

```diff
@@ -718,8 +720,142 @@ def scp_between_remotes(src, dst, port, s_passwd, d_passwd, s_name, d_name,
                 fr" {s_name}@\[{src}\]:{quote_path(s_path)} {d_name}@\[{dst}\]"
                 fr":{shlex.quote(d_path)}")
     password_list = [s_passwd, d_passwd]
-    return remote_scp(command, password_list,
-                      log_filename, log_function, timeout)
+    return remote_copy(command, password_list,
+                       log_filename, log_function, timeout)
+
+
+def rsync_to_remote(host, port, username, password, local_path, remote_path,
+                    limit="", log_filename=None, log_function=None,
+                    timeout=600, interface=None, directory=True):
+    """
+    Copy files to a remote host (guest) through rsync.
+
+    :param host: Hostname or IP address
+    :param port: Port
+    :param username: Username (if required)
+    :param password: Password (if required)
+    :param local_path: Path on the local machine where we are copying from
+    :param remote_path: Path on the remote machine where we are copying to
+    :param limit: Speed limit of file transfer.
+    :param log_filename: If specified, log all output to this file
+    :param log_function: If specified, log all output using this function
+    :param timeout: The time duration (in seconds) to wait for the transfer
+                    to complete.
+    :param interface: The interface the neighbours attach to (only use when using
+                      ipv6 linklocal address).
+    :param directory: True to copy recursively if the directory to rsync
```

Comment:
this seems more important then `interface`, shouldn't it go first?

Reply:
We tend to avoid changing the order of parameters for users that don't rely on keywords and this is simply an adaptation of the `scp_from_remote` which has the same order (and likely user code assuming this).